config/config.go

```diff
@@ -170,4 +170,13 @@ type Config interface {
 	GetAddSpanCountToRoot() bool
 
 	GetCacheOverrunStrategy() string
+
+	GetConfigMetadata() []ConfigMetadata
+}
+
+type ConfigMetadata struct {
+	Type     string `json:"type"`
+	ID       string `json:"id"`
```

Comment:
I\m not sure ID is clear that it's going to return the path to the file, should we use `path` instead?

Reply:
I chose not to name it path or filename because I have longterm plans to allow configuration in other ways besides files that have a discernable path (for example, a shared configuration in Redis).